analyzer/src/funTest/kotlin/managers/Extensions.kt

```diff
@@ -55,18 +56,20 @@ fun PackageManager.resolveSingleProject(definitionFile: File, resolveScopes: Boo
 fun PackageManagerResult.resolveScopes(projectResult: ProjectAnalyzerResult): ProjectAnalyzerResult {
     val resolvedProject = projectResult.project.withResolvedScopes(dependencyGraph)
 
-    // The result must contain packages for all the dependencies declared by the project; otherwise, the
-    // check in ProjectAnalyzerResult.init fails. When using a shared dependency graph, the set of packages
-    // is typically empty, so it has to be populated manually from the subset of shared packages that are
-    // referenced from this project. If there is a single project only, use all packages; this handles corner
-    // cases with package managers producing packages not assigned to project scopes.
+    // When using a shared dependency graph, the set of packages is typically empty, so it has to be populated manually
+    // from the subset of shared packages that are referenced from this project. If there is a single project only, use
+    // all packages; this handles corner cases with package managers producing packages not assigned to project scopes.
     val packages = projectResult.packages.takeUnless { it.isEmpty() }
         ?: if (projectResults.size > 1) resolvedProject.filterReferencedPackages(sharedPackages) else sharedPackages
     return projectResult.copy(project = resolvedProject, packages = packages.toSortedSet())
 }
 
 /**
  * Return only those packages from the given set of [allPackages] that are referenced by this [Project].
+ * NOTE: The project is known to use the scopes structure for storing its dependencies; therefore, a
+ * [DependencyTreeNavigator] can be used to access this information.
  */
-private fun Project.filterReferencedPackages(allPackages: Set<Package>): SortedSet<Package> =
-    collectDependencies().run { allPackages.filter { it.id in this }.toSortedSet() }
+private fun Project.filterReferencedPackages(allPackages: Set<Package>): SortedSet<Package> {
+    val projectDependencies = DependencyTreeNavigator.projectDependencies(this)
+    return projectDependencies.run { allPackages.filter { it.id in this }.toSortedSet() }
```

Comment:
`run` is not needed anymore: `return allPackages.filter { it.id in projectDependencies }.toSortedSet()`